public-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -236,4 +247,112 @@ public ConceptIds classifyConceptIds(Set<Long> conceptIds) {
+            if(standardConceptFilter == null || standardConceptFilter == StandardConceptFilter.STANDARD_OR_CODE_ID_MATCH){
+                standardConceptFilter = StandardConceptFilter.STANDARD_CONCEPTS;
+            }
+        }else{
+            if(standardConceptFilter == null){
+                standardConceptFilter = StandardConceptFilter.STANDARD_OR_CODE_ID_MATCH;
+            }
+        }
+
+        String domainId = null;
+        if (searchConceptsRequest.getDomain() != null) {
+            domainId = CommonStorageEnums.domainToDomainId(searchConceptsRequest.getDomain());
+        }
+
+        StandardConceptFilterVal convertedConceptFilter = StandardConceptFilterVal.valueOf(standardConceptFilter.name());
+
+        Slice<DbConcept> concepts = null;
+        int measurementTests = 1;
+        int measurementOrders = 1;
+
+        if (domainId != null && domainId.equals("Measurement")) {
+            if (searchConceptsRequest.getMeasurementTests() != null) {
+                measurementTests = searchConceptsRequest.getMeasurementTests();
+            }
+            if (searchConceptsRequest.getMeasurementOrders() != null) {
+                measurementOrders = searchConceptsRequest.getMeasurementOrders();
+            }
+        }
+        concepts = searchConcepts(searchConceptsRequest.getQuery(), convertedConceptFilter, drugConcepts,
+                searchConceptsRequest.getVocabularyIds(), domainId, maxResults, minCount,
+                (searchConceptsRequest.getPageNumber() == null) ? 0 : searchConceptsRequest.getPageNumber(), measurementTests, measurementOrders);
+
+        ConceptListResponse response = new ConceptListResponse();
+
+        for(DbConcept con : concepts.getContent()){
+            String conceptCode = con.getConceptCode();
+            String conceptId = String.valueOf(con.getConceptId());
+
+            if((con.getStandardConcept() == null || !con.getStandardConcept().equals("S") ) && (searchConceptsRequest.getQuery().equals(conceptCode) || searchConceptsRequest.getQuery().equals(conceptId))){
+                response.setStandardConcepts(getStandardConcepts(con.getConceptId()));
+                response.setSourceOfStandardConcepts(con.getConceptId());
+            }
+
+            if(!Strings.isNullOrEmpty(searchConceptsRequest.getQuery()) && (searchConceptsRequest.getQuery().equals(conceptCode) || searchConceptsRequest.getQuery().equals(conceptId))) {
+                response.setMatchType(conceptCode.equals(searchConceptsRequest.getQuery()) ? MatchType.CODE : MatchType.ID );
+                response.setMatchedConceptName(con.getConceptName());
+            }
+        }
+
+        if(response.getMatchType() == null && response.getStandardConcepts() == null){
+            response.setMatchType(MatchType.NAME);
+        }
+
+        List<Concept> conceptList = new ArrayList<>();
+
+        if (concepts != null) {
```

Comment:
Variable `concepts` will always be true. Need to check context.
```suggestion
        if (concepts.getContent() != null) {
```

Reply:
Changed this.